### PR TITLE
fix: harpoon telescope integration

### PR DESCRIPTION
**Pull request by hmajid2301**
At the moment if your config doesn't specify `enableTelescope` in the harpoon options you get the following:

Harpoon config:
```
      harpoon = {
        enable = true;
        tmuxAutocloseWindows = true;
        keymaps = {
          addFile = "<leader>ha";
          toggleQuickMenu = "<leader>ht";
          navNext = "<leader>hn";
          navPrev = "<leader>hp";
        };
      };
```

Error:
```bash
error:
       … while evaluating a branch condition

         at /nix/store/4smdp3v5a9bcg1hpdhl2hs70rq385ykq-source/lib/lists.nix:57:9:

           56|       fold' = n:
           57|         if n == len
             |         ^
           58|         then nul

       … while calling the 'length' builtin

         at /nix/store/4smdp3v5a9bcg1hpdhl2hs70rq385ykq-source/lib/lists.nix:55:13:

           54|     let
           55|       len = length list;
             |             ^
           56|       fold' = n:

       (stack trace truncated; use '--show-trace' to show the full trace)

       error:
       Failed assertions:
       - Nixvim: The harpoon telescope integration needs telescope to function as intended
```

I think we should default it to `false` and make users turn it on.

### CODE REVIEW

**Comment by traxys** [comment]
This seems strange, because mkEnableOption should already default to false

**Comment by hmajid2301** [comment]
it could be the assertion logic is incorrect ?

**Comment by hmajid2301** [comment]
yeh i think the assertion logic shouldve been reversed my bad :sweat: 

if harpoon extension enabled -> telescope is enabled

**Comment by GaetanLepage** [comment]
Why have you removed the `keymapsSilent` option ?

**Comment by hmajid2301** [comment]
my bad fixed now!

File: plugins/utils/harpoon.nix [hunk 1]
```diff
@@ -27,7 +27,7 @@ in {
 
       package = helpers.mkPackageOption "harpoon" pkgs.vimPlugins.harpoon;
 
-      enableTelescope = mkEnableOption "Enable telescope integration";
+      enableTelescope = mkEnableOption "enable telescope integration";
```
Comment (GaetanLepage) [inline plugins/utils/harpoon.nix:30]:
```suggestion
      enableTelescope = mkEnableOption "telescope integration";
```

Comment (GaetanLepage) [reply plugins/utils/harpoon.nix:30]: `mkEnableOption` already prefixes your provided description with `Whether to enable `.

**Comment by hmajid2301** [comment]
good to know! For next time.

**Review by GaetanLepage — APPROVED** [review]
Well done !

**Pull request merged**